### PR TITLE
Catch db_engine_spec.get_function_names exceptions

### PR DESCRIPTION
**Pull request by bkyryliuk**
### CATEGORY

Choose one

- [x] Bug Fix

### SUMMARY
Short term fix for the https://github.com/apache/incubator-superset/issues/9678
Databases api fetches all database function names, it requires the databases to be online and return the values successfully to be able to fetch the list of the databases.
If it fails sqllab is not able to get the list of the databases and becomes disfunctional

### TEST PLAN
tested on the dropbox staging environment, provided bad DB url and made sure that sqllab works.

### ADDITIONAL INFORMATION

- [x] Has associated issue:

### REVIEWERS
@villebro @dpgaspar

### CODE REVIEW

**Comment by bkyryliuk** [comment]
github checks somehow did not rerun when the commit was amended, still shows old code here:

image

**Review by dpgaspar — APPROVED** [review]

**Review by villebro — APPROVED** [review]

**Review by etr2460 — APPROVED** [review]

**Comment by bkyryliuk** [comment]
@dpgaspar, @villebro or @etr2460 could someone merge it, I am still waiting on getting the merge rights :(

**Comment by dpgaspar** [comment]
merging, sorry for the delay

**Pull request merged**
